crates/stc_ts_file_analyzer/src/analyzer/scope/vars.rs

```diff
@@ -160,20 +172,58 @@ impl Analyzer<'_, '_> {
                 let is_typed = type_ann.is_some();
                 let mut type_ann = type_ann.or(default);
                 type_ann.freeze();
-
+                println!("@@@@@@@@@@@ Assign {:?}", p);
                 let mut right = p
                     .right
                     .validate_with_args(self, (TypeOfMode::RValue, None, type_ann.as_ref().or(ty.as_ref())))
                     .report(&mut self.storage)
                     .unwrap_or_else(|| Type::any(span, Default::default()));
 
+                println!("@@@@@@@@@@@ Assign {:?}", p.left);
+                println!("@@@@@@@@@@@ Assign {:?}", p.right);
+                println!("@@@@@@@@@@@ Assign {:?}", is_typed);
+                println!("@@@@@@@@@@@ Assign {:?}", type_ann);
+
                 if self.ctx.is_fn_param && type_ann.is_none() {
                     // If the declaration includes an initializer expression (which is permitted
                     // only when the parameter list occurs in conjunction with a
                     // function body), the parameter type is the widened form (section
                     // 3.11) of the type of the initializer expression.
+                    match &*p.left {
+                        RPat::Array(p_left) => {
+                            println!("tuple@@@!!! {:?}", right);
+
+                            right = right.fold_with(&mut Widen { tuple_to_array: false });
```

Comment:
The direction looks fine, but isn't this logic common for all cases?
I think this can be moved to up of the match expression

Reply:
In my opinion, `right = right.fold_with(&mut Widen { tuple_to_array: false });` is not common logic.

` right = right.fold_with(&mut Widen { tuple_to_array: ture });` is the existing logic, 
If I do this In situations like `function f03([x, y, z] = [1])`, the type is inferred as `number[]` instead of `[number]`.

So if left is an array, and `is_fn_param=true`, I want to do the logic `right = right.fold_with(&mut Widen { tuple_to_array: false });`,

Reply:
Ah, my bad. I missed the `tuple_to_array`